NewHorizons/Components/StarLightController.cs

```diff
@@ -78,13 +93,14 @@ public void Update()
                 material.SetFloat(SunIntensity, intensity);
             }
 
+            // Player is always at 0,0,0 more or less so if they arent using the map camera then wtv
+            var origin = Vector3.zero;
+
             foreach (var star in _stars)
             {
                 if (star == null) continue;
                 if (!(star.gameObject.activeSelf && star.gameObject.activeInHierarchy)) continue;
 
-                // Player is always at 0,0,0 more or less so if they arent using the map camera then wtv
-                var origin = Vector3.zero;
                 if (PlayerState.InMapView())
```

Comment:
Since you brought the origin part out of this for loop, bring this if statement with it